### PR TITLE
DAOS-7821 gha: Remove pylint from checks.

### PR DESCRIPTION
**Pull request by ashleypittman**
This should all be handled via github actions now, so remove
from Jenkins.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
Should be reviewed in combination with https://github.com/daos-stack/code_review/pull/31  I think these PRs are correct for master and disable all codespell and pylint checks from Jenkins but will have the side effect of also removing them for other branches (both release and feature).  I can't think of a way around this without compromising the master tree however, perhaps the code-review change could check for the existence of utils/cq/daos_pylint.py?

The github actions checks for codespell and pylint should be feature compatible with Jenkins now although there are slight differences because of various settings and the way issues are reported.

**Review by jolivier23 — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -337,6 +337,7 @@ pipeline {
                     steps {
                         checkPatch user: GITHUB_USER_USR,
                                    password: GITHUB_USER_PSW,
+                                   branch: 'amd/no-pylint',
```
Comment (brianjmurrell) [inline Jenkinsfile:340]: This shouldn't land, correct?  Will -1 just to ensure it doesn't happen accidentally.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (ashleypittman) [reply Jenkinsfile]: Removed.

**Review by jolivier23 — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Comment by brianjmurrell** [comment]
> ashleypittman force-pushed the amd/no-pylint branch from f261d19 to 5eacf0f 

Please try to remember not to force push once reviewing has started.

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**